libsql-server/src/connection/connection_manager.rs

```diff
@@ -126,12 +127,20 @@ impl ManagedConnectionWalWrapper {
             // - we tried to acquire the lock during the previous iteration, but the underlying
             // method returned an error and we had to retry immediately, by re-entering this
             // function.
+            let new_token = self.manager.sync_token.load(Ordering::SeqCst);
+            if new_token != sync_token {
+                return Err(rusqlite::ffi::Error {
+                    code: ErrorCode::DatabaseBusy,
+                    extended_code: 517, // stale read
+                });
+            }
             if current.map_or(true, |(id, _, _)| id != self.id) && !enqueued {
+                sync_token = new_token;
```

Comment:
Is it ok not to do CompareAndSet here? There's never a race in which:
1. new_token == sync_token
2. sync_token is changed by someone else
3. we override sync_token with new_token
?